backend/clubs/views.py

```diff
@@ -3081,74 +3080,54 @@ def get_queryset(self):
-        return ClubFairBooth.objects.filter(
-            club__code=self.kwargs["club__code"]
-        ).select_related("club")
-
+    get:
+    Get club booths corresponding to club code
 
-class ClubBoothsSneakyAPIView(generics.ListAPIView):
-    """
-    Return
+    post:
+    Create or update a club booth
     """
 
-    serializer_class = DummyBoothSerializer
+    lookup_field = "club__code"
+    queryset = ClubFairBooth.objects.all()
+    serializer_class = ClubBoothSerializer
     permission_classes = [IsAuthenticated]
+    http_methods_names = ["get", "post"]
 
-    def get_queryset(self):
-        kwargs = self.kwargs
-        dum = ClubFairBooth.objects.get_or_create(
-            name="Dummy",
-            club=Club.objects.all().first(),
-            lat=0.0,
-            long=0.0,
-            start_time=datetime.datetime.now(),
-            end_time=datetime.datetime.now(),
-        )
-        try:
-            name = kwargs["name"]
-            code = kwargs["club__code"]
-            image_url = kwargs["image_url"]
-            lat = kwargs["lat"]
-            lon = kwargs["lon"]
-            subtitle = kwargs["subtitle"]
-            start_time = parse(kwargs["start_time"])
-            end_time = parse(kwargs["end_time"])
-
-            club = Club.objects.filter(code=code).first()
-            if club:
-                mem = find_membership_helper(self.request.user, club)
-                if (
-                    mem
-                    and mem.role > 10
-                    and self.request.user.email != "jongmin@sas.upenn.edu"
-                ):
-                    return dum
-
-            booth = ClubFairBooth.objects.filter(club__code=code).first()
-            if not booth:
-                booth = ClubFairBooth(club=club)
-            booth.name = name
-            booth.image_url = image_url
-            booth.lat = float(lat)
-            booth.lon = float(lon)
-            booth.subtitle = subtitle
-            booth.start_time = start_time
-            booth.end_time = end_time
-            if end_time < start_time:
-                return dum
-            booth.save()
-        except (Exception,):
-            return dum
-        return dum
+    @action(detail=False, methods=["post"])
+    def create_or_update(self, request, *args, **kwargs):
+        name = kwargs["name"]
+        code = kwargs["club__code"]
+        image_url = kwargs["image_url"]
+        lat = kwargs["lat"]
+        lon = kwargs["lon"]
+        subtitle = kwargs["subtitle"]
+        start_time = parse(kwargs["start_time"])
+        end_time = parse(kwargs["end_time"])
+        booth = ClubFairBooth.objects.filter(club__code=code).first()
+        club = Club.objects.filter(code=code).first()
+
+        if club:
+            mem = find_membership_helper(self.request.user, club)
+            if (
+                mem
+                and mem.role > 10
+                and self.request.user.email != "jongmin@sas.upenn.edu"
```

Comment:
Why

Reply:
only allowing officer+ to edit/add things + jong min for testing